actors/builtin/miner/policy.go

```diff
@@ -28,8 +32,8 @@ func init() {
 	}
 }
 
-// The maximum number of sectors that a miner can have simultaneously active.
-// This also bounds the number of faults that can be declared, etc.
+// Filecoin Parameter: The maximum number of sectors that a miner can have (and implicitly the maximum number of faults, recoveries, etc.) 
+// Motivation: This guarantees that actors operations per miner are bounded. Also, this reflects the limit of sectors that can be proven in a day via WindowPoSt.
 // TODO raise this number, carefully
 // https://github.com/filecoin-project/specs-actors/issues/470
 const SectorsMax = 32 << 20 // PARAM_FINISH
```

Comment:
Problems that we found:
- Previously this was referred to as "The maximum number of sectors that a miner can have simultaneously active.", we believe this is either incorrect or subject to complex edge cases (e.g. a miner that has 32<<20 sectors active and 1 sector faulty which then recovers the faulty sector, now has 32<<20+1 active sectors). Can @ZenGround0 or @anorth confirm this?
- `SectorsMax` is specific to a particular `RegisteredProof`. For example, NSE will have a much smaller `SectorsMax`. Why? One of the constrains that sets the `SectorsMax` is the number of sectors that can be proven in 24 hours via a WindowPost. If we have NSE, the number of sectors that can be proven in a WindowPoSt is reduced, hence the SectorsMax chosen for SDR will be much larger.

Reply:
Sounds right but you'll need @anorth or @Stebalien to confirm

Reply:
what does it mean for them to be "not yet compacted out"?
Does this imply that the maximum number of sectors can be higher than 32 << 20?

Reply:
> Does this imply that the maximum number of sectors can be higher than 32 << 20?

No I understand the above comment to mean that this max covers all sectors, active, faulty and terminated but not yet cleaned up

Reply:
>  Does this imply that the maximum number of sectors can be higher than 32 << 20?

Lower.

When we terminate a sector, we just mark it as terminated and pay any fees. However, we don't actually cleanup the state (sector info) or remove it from the deadline/partition till the miner calls a "compact" method on the deadline/partition. When the miner calls "compact", re-create the deadline from all _live_ sectors in the partition, and remove any state related to old sectors.

I say "lower" because `32<<20` is the max number of sectors _including_ these terminated-but-not-yet-compacted sectors.

Reply:
This is resolved, to me - feel free to re-open in case you are not happy with the wording I used.